src/System.Linq/src/System/Linq/Iterator.cs

```diff
@@ -67,5 +67,50 @@ void IEnumerator.Reset()
                 throw Error.NotSupported();
             }
         }
+
+        internal abstract class ManualIterator<TSource> : IEnumerable<TSource>, IEnumerator<TSource>
```

Comment:
I didn't add XML docs for this yet, but the name was chosen because the subclassing iterator manually supplies an implementation of `Current`.